internal/xsql/rows.go

```diff
@@ -48,6 +49,23 @@ func (r *rows) Columns() []string {
 	return cs
 }
 
+// NOTE: Need to optimize somehow? This might take O(|r.Columns()|^2)
+// each time (*Rows).ColumnTypes be called
+// https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/database/sql/sql.go;l=3101
+func (r *rows) ColumnTypeDatabaseTypeName(index int) string {
+	r.nextSet.Do(func() {
+		r.result.NextResultSet(context.Background())
+	})
+
+	var i int
+	yqlTypes := make([]string, r.result.CurrentResultSet().ColumnCount())
```

Comment:
it must be a some cache field of rows. for second call

Reply:
Or at now it look like overengenering, bc making slice and call .Yql() is not free